client/client-multi/client-cpp/CMakeLists.txt

```diff
@@ -204,6 +204,10 @@ if(NOT MSVC)
    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -ggdb")
 endif()
 
+# "no-unused-private-field" is added since in most of cases it is false positive
+list(APPEND KAA_COMPILE_OPTIONS
+                -Wno-unused-private-field)
```

Comment:
I do not "remove any mention of IKaaClientContext from ProfileManager.hpp and had edited KaaClient.cpp". I just added flag to do not see this warning any more. 

I do not see mentions of "IKaaClientContext and ProfileManager.hpp" while compiling this module:
Building CXX object CMakeFiles/kaatest.dir/Users/vasiliy/Works/CyberVision/Internal/kaa/git-repos/kaa-vnerozin/client/client-multi/client-cpp/impl/logging/DefaultLogger.cpp.o